main/modes/timer/modeTimer.c

```diff
@@ -0,0 +1,286 @@
+    int64_t startTime;
+} timerMode_t;
+
+//==============================================================================
+// Function Prototypes
+//==============================================================================
+
+static void timerEnterMode(void);
+static void timerExitMode(void);
+static void timerMainLoop(int64_t elapsedUs);
+
+//==============================================================================
+// Strings
+//==============================================================================
+
+static const char timerName[]              = "Timer";
+static const char hourglassFrameFmt[]      = "hourglass_%02" PRIu8 ".wsg";
+static const char minutesSecondsFmt[]      = "%" PRIu8 ":%02" PRIu8 ".";
+static const char hoursMinutesSecondsFmt[] = "%" PRIu64 ":%02" PRIu8 ":%02" PRIu8 ".";
+static const char millisFmt[]              = "%03" PRIu16;
+// static const char minutesSecondsBg[] = "88:88.";
+// static const char millisBg[] = "888";
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+swadgeMode_t timerMode = {
+    .modeName                 = timerName,
+    .wifiMode                 = NO_WIFI,
+    .overrideUsb              = false,
+    .usesAccelerometer        = true,
+    .usesThermometer          = false,
+    .overrideSelectBtn        = false,
+    .fnEnterMode              = timerEnterMode,
+    .fnExitMode               = timerExitMode,
+    .fnMainLoop               = timerMainLoop,
+    .fnAudioCallback          = NULL,
+    .fnBackgroundDrawCallback = NULL,
+    .fnEspNowRecvCb           = NULL,
+    .fnEspNowSendCb           = NULL,
+    .fnAdvancedUSB            = NULL,
+};
+
+static timerMode_t* timerData = NULL;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+static void timerEnterMode(void)
+{
+    timerData = calloc(1, sizeof(timerMode_t));
+
+    for (uint8_t i = 0; i < HOURGLASS_FRAMES; i++)
+    {
+        char wsgName[18];
+        snprintf(wsgName, sizeof(wsgName), hourglassFrameFmt, i);
+        loadWsg(wsgName, &timerData->hourglassFrames[i], false);
+    }
+
+    loadFont("ibm_vga8.font", &timerData->textFont, false);
+    loadFont("seven_segment.font", &timerData->numberFont, false);
+
+    // 100FPS? Sure?
+    setFrameRateUs(1000000 / 100);
+
+    // Default to 30s
+    timerData->countdownTime = 30 * 1000000;
+    timerData->timerState    = STOPPED;
+}
+
+static void timerExitMode(void)
+{
+    for (uint8_t i = 0; i < HOURGLASS_FRAMES; i++)
+    {
+        freeWsg(&timerData->hourglassFrames[i]);
+    }
+
+    freeFont(&timerData->textFont);
+    freeFont(&timerData->numberFont);
+
+    free(timerData);
+    timerData = NULL;
+}
+
+static void timerMainLoop(int64_t elapsedUs)
+{
+    int64_t now = esp_timer_get_time();
```

Comment:
Why are you making your own time calls? `elapsedUs` should have you covered, no?

Reply:
That does get funky if you open the quick settings menu, so I wanted to just use the total time and not have to worry about losing any time anywhere.

Reply:
Fair enough.